From 89293e44e0d022463b03a92a30b3790d4569bd50 Mon Sep 17 00:00:00 2001
From: Mark Andrews <marka@isc.org>
Date: Thu, 20 Feb 2014 23:04:54 +1100
Subject: [PATCH] 3752.   [bug]           Address potential REQUIRE failure if 
                        DNS_STYLEFLAG_COMMENTDATA is set when printing out    
                     a rdataset.

(cherry picked from commit 86856f4f3069bb2d75851b56401ffde18f41198f)
---
 lib/dns/masterdump.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/dns/masterdump.c b/lib/dns/masterdump.c
index 80fcd4c12d..58dcd3de3e 100644
--- a/lib/dns/masterdump.c
+++ b/lib/dns/masterdump.c
@@ -451,7 +451,7 @@ rdataset_totext(dns_rdataset_t *rdataset,
 		 * Comment?
 		 */
 		if ((ctx->style.flags & DNS_STYLEFLAG_COMMENTDATA) != 0)
-			isc_buffer_putstr(target, ";");
+			RETERR(str_totext(";", target));
 
 		/*
 		 * Owner name.
-- 
2.14.5

